sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionPump.cs

```diff
@@ -90,36 +97,40 @@ internal class PartitionPump
         ///
         private Task RunningTask { get; set; }
 
-        /// <summary>
-        ///   The reason why the processing for the associated partition is being stopped.  This member is only used
-        ///   in case of failure.  Shutdown and OwnershipLost close reasons will be specified by the event processor.
-        /// </summary>
-        ///
-        private CloseReason CloseReason { get; set; }
-
         /// <summary>
         ///   Initializes a new instance of the <see cref="PartitionPump"/> class.
         /// </summary>
         ///
-        /// <param name="eventProcessor">The <see cref="EventProcessorClient" /> that owns this instance.</param>
         /// <param name="connection">The <see cref="EventHubConnection" /> connection to use for communication with the Event Hubs service.</param>
         /// <param name="consumerGroup">The name of the consumer group this partition pump is associated with.  Events are read in the context of this group.</param>
         /// <param name="partitionContext">The context of the Event Hub partition this partition pump is associated with.  Events will be read only from this partition.</param>
         /// <param name="startingPosition">The position within the partition where the pump should begin reading events.</param>
+        /// <param name="processEventAsync">Responsible for processing events received from the Event Hubs service.</param>
+        /// <param name="updateCheckpointAsync">Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.</param>
         /// <param name="options">The set of options to use for this partition pump.</param>
         ///
-        internal PartitionPump(EventProcessorClient eventProcessor,
-                               EventHubConnection connection,
+        internal PartitionPump(EventHubConnection connection,
                                string consumerGroup,
                                PartitionContext partitionContext,
```

Comment:
You may need to make this generic and treat the partition context as opaque, assuming this will be called by the base class.    You can always grab the partition id and any other information separately if you need them.

Reply:
Will be done on the next PR (Event Processor Base-related).